pipeline/load.py

```diff
@@ -0,0 +1,288 @@
+    Returns a connection if successful.
+    """
+
+    logging.info("Connecting to the database.")
+
+    try:
+        connection = psycopg2.connect(
+            host=environ["DB_HOST"],
+            port=environ["DB_PORT"],
+            user=environ["DB_USER"],
+            password=environ["DB_PASSWORD"],
+            database=environ["DB_NAME"]
+        )
+        return connection
+    except psycopg2.OperationalError:
+        logging.warning("The database %s doesn't exist", environ["DB_NAME"])
+        return None
+
+
+def get_cursor(connection: psycopg2.extensions.connection) -> psycopg2.extensions.cursor:
+    """
+    Returns a the cursor for querying
+    the database from a connection.
+    """
+    try:
+        return connection.cursor()
+    except:
+        logging.warning("Unable to retrieve cursor for connection")
+        return None
+
+
+def get_id_from_country(country_name: str, cursor: psycopg2.extensions.cursor) -> int:
+    """
+    Retrieves the country_id of a 
+    given country from the database.
+    Returns -1 if it cannot be found.
+    """
+    search_query = f"SELECT country_id FROM country WHERE country_name = '{
+        country_name}';"
+    cursor.execute(search_query)
+    country_id = cursor.fetchone()
+    if not country_id:
+        logging.info("Unable to find id for %s", country_name)
+        return -1
+    return country_id[0]
+
+
+def get_id_from_artist(artist_name: str, cursor: psycopg2.extensions.cursor) -> int:
+    """
+    Retrieves the artist id of a 
+    given artist from the database.
+    Returns -1 if it cannot be found.
+    """
+    search_query = "SELECT artist_id FROM artist WHERE artist_name = %s;"
+    cursor.execute(search_query, (artist_name,))
+    artist_id = cursor.fetchone()
+    if not artist_id:
+        logging.info("Unable to find id for %s", artist_name)
+        return -1
+    return artist_id[0]
+
+
+def get_id_from_release_type(release_type: str, cursor: psycopg2.extensions.cursor) -> int:
+    """
+    Retrieves the release type id of a 
+    given release type.
+    Returns -1 if it cannot be found.
+    """
+    search_query = "SELECT type_id FROM type WHERE type_name = %s;"
+    cursor.execute(search_query, (release_type.lower(),))
+    type_id = cursor.fetchone()
+    if not type_id:
+        logging.info("Unable to find id for %s", release_type)
+        return -1
+    return type_id[0]
+
+
+def insert_country(country_name: str, cursor: psycopg2.extensions.cursor) -> None:
+    """
+    Inserts a country into the database if it doesn't already exist.
+    Logs whether the country was added or already exists.
+    """
+    try:
+        check_query = "SELECT EXISTS (SELECT 1 FROM country WHERE country_name = %s);"
+        cursor.execute(check_query, (country_name,))
+        exists = cursor.fetchone()[0]
+
+        if exists:
+            logging.info(
+                "Country '%s' already exists in the database.", country_name)
```

Comment:
Do we want to log this every time? Surely pretty quickly every country will exist in the database and we will just constantly spamming our logs with this?